roles/image-cache/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Create image-cache directory
+  become: true
+  file:
+    path: "{{ images_dir }}"
+    state: directory
+
+- name: Check if OS image exists
+  stat:
+    path: "{{ images_dir }}/{{ image_name }}"
+  register: image_path
+
+- name: Download and modify OS image if it doesn't exist
+  when: "not image_path.stat.exists"
+  become: true
+  block:
+  - name: Download OS image
+    get_url:
+      url: "{{ image_url }}"
+      dest: "{{ images_dir }}/{{ image_name }}"
+      checksum: "{{ image_checksum }}"
+
+  - name: Generate OS image checksum file
+    shell: |
+      sha256sum "{{ image_name }}" > "{{ image_name }}.sha256"
+    args:
+      chdir: "{{ images_dir }}"
+
+  - ansible.builtin.debug:
+
+  - ansible.builtin.debug:
+      msg: "Downloaded {{ images_dir }}/{{ image_name }}"
+
+  - name: Copy file OS image file
+    ansible.builtin.copy:
+      src: "{{ images_dir }}/{{ image_name }}"
+      dest: "{{ images_dir }}/{{ metal3_image_name }}"
+
+  # Use "opensuse:root" credentials for the dev user.
+  # Generate another password by running "openssl passwd -6 -salt SalTsaLt PASSWORD"
+  # and replacing the "passwd" entry below (keep in mind some characters must be escaped e.g. "/", "$").
+  - name: Modify OS image cloud-init config
+    shell: |
+      virt-edit -a "{{ metal3_image_name }}" /etc/cloud/cloud.cfg -e 's/datasource_list.*/datasource_list: [ ConfigDrive, OpenStack, None ]/'
+      virt-edit -a "{{ metal3_image_name }}" /etc/cloud/cloud.cfg -e 's/(\s*)lock_passwd.*/\1lock_passwd: False/'
+      virt-edit -a "{{ metal3_image_name }}" /etc/cloud/cloud.cfg -e 's/(\s*)(name: opensuse)/\1\2\n\1passwd: \$6\$SalTsaLt\$4U6ktRhvpSUKdG2XHWmsxlvykFSJRnMyOLYPcWLoCuj3oHB4NaE9Is0wXf4guihZvmSBTairoUvh6s5.JoJED\//'
```

Comment:
Can we move this password to an arg in the main.yml?

Reply:
Yes this was copied from the [existing role](https://github.com/suse-edge/metal3-demo/blob/main/roles/diskimage_builder/tasks/build_metal3_images.yml) but I can certainly move the password to a variable

Reply:
I will do this as a follow-up PR since this is copied from an existing role.